src/server/pfs/s3/object.go

```diff
@@ -5,13 +5,34 @@ import (
 	"io"
 	"net/http"
 	"strings"
+	"time"
 
 	"github.com/gogo/protobuf/types"
 	"github.com/pachyderm/pachyderm/v2/src/internal/errutil"
+	pfsClient "github.com/pachyderm/pachyderm/v2/src/pfs"
 	pfsServer "github.com/pachyderm/pachyderm/v2/src/server/pfs"
 	"github.com/pachyderm/s2"
 )
 
+func permissiveTimestampFromProto(fileInfo *pfsClient.FileInfo) (time.Time, error) {
+	// fileInfo.Committed can be nil in the case of reading from an open
+	// commit. In that case, it's better to list the file with no known
+	// timestamp than to return an error.
+
+	// TODO: it would be less wrong to use the commit start time than to return
```

Comment:
I think this is too late to do anything about this, though, it would need to be a change in the iteration in pfs/server/source.go  
It doesn't look like we currently use that field to indicate whether the source commit is finished, so it would probably be okay to use the start time there